lang/src/org/partiql/lang/typesystem/builtin/types/nonprimitive/CharType.kt

```diff
@@ -0,0 +1,37 @@
+package org.partiql.lang.typesystem.builtin.types.nonprimitive
+
+import org.partiql.lang.eval.ExprValue
+import org.partiql.lang.eval.ExprValueType
+import org.partiql.lang.typesystem.builtin.types.primitive.IntType
+import org.partiql.lang.typesystem.builtin.types.valueFactory
+import org.partiql.lang.typesystem.interfaces.type.BuiltInType
+import org.partiql.lang.typesystem.interfaces.type.ParametricType
+import org.partiql.lang.typesystem.interfaces.type.SqlType
+import org.partiql.lang.typesystem.interfaces.type.TypeParameters
+import org.partiql.lang.util.asIonInt
+
+object CharType : BuiltInType(), ParametricType {
```

Comment:
For this and others that apply: let's have a decent `kdoc` for the types with examples.

Reply:
Do we want to write documentations to explain what each type does?

Reply:
> Do we want to write documentations (as separate files) to explain what each type does?
That can also be done, but at its very least we need some sort of documentation in the code.